### PR TITLE
Only allow Maps as variable for InputObjects

### PR DESCRIPTION
**Pull request by bbakerman**
See #739 

Currently this is just the test - no fix - so it should fail

### CODE REVIEW

**Comment by bbakerman** [comment]
I updated this to be graphql spec compliant as we discussed in the gitter room.

File: src/main/java/graphql/execution/ValuesResolver.java [hunk 1]
```diff
@@ -145,19 +146,23 @@ private Object coerceValue(VariableDefinition variableDefinition, GraphQLType gr
...
-        } else if (graphQLType instanceof GraphQLInputObjectType && value instanceof Map) {
-            //noinspection unchecked
-            return coerceValueForInputObjectType(variableDefinition, (GraphQLInputObjectType) graphQLType, (Map<String, Object>) value);
         } else if (graphQLType instanceof GraphQLInputObjectType) {
-            return value;
+            if (value instanceof Map) {
+                //noinspection unchecked
+                return coerceValueForInputObjectType(variableDefinition, (GraphQLInputObjectType) graphQLType, (Map<String, Object>) value);
+            } else {
+                throw new CoercingParseValueException("Variables for GraphQLInputObjectType must be an instance of a Map according to the graphql specification.  The offending object was a " + value.getClass().getName());
```
Comment (RusticFlare) [inline src/main/java/graphql/execution/ValuesResolver.java]: Sorry for being picky but there's a double space after the full stop 😊

Comment (bbakerman) [reply src/main/java/graphql/execution/ValuesResolver.java]: Sorry I am old  and learnt to type  on mechanical typewriters and Mrs Cleary taught us that double spaces after a period are how you do it.  Its burnt into my muscle memory. ;)

**Comment by andimarek** [comment]
I updated the title to make it more clear what this PR actually does.

**Pull request merged**